libminifi/src/FlowController.cpp

```diff
@@ -173,9 +173,9 @@ void FlowController::initializePaths(const std::string &adjustedFilename) {
 
 FlowController::~FlowController() {
   stop(true);
+  c2_agent_->stop();
```

Comment:
Are you sure we can  avoid the nullcheck here?
I don't think we have C2 agent in case c2 is not configured in the minifi.properties file

Reply:
No we cannot avoid null check here. Fixed it in latest commit